src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt

```diff
@@ -0,0 +1,33 @@
+package pt.up.fe.ni.website.backend.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import pt.up.fe.ni.website.backend.model.Activity
+import pt.up.fe.ni.website.backend.repository.ActivityRepository
+
+@Service
+abstract class ActivityService<T : Activity>(
+    private val repository: ActivityRepository<T>,
+    private val accountService: AccountService
+) {
+    fun getActivityById(id: Long): T = repository.findByIdOrNull(id)
+        ?: throw NoSuchElementException(ErrorMessages.activityNotFound(id))
```

Comment:
this is kind of weird because, when using a ProjectService, we'll have `getActivityById` and `getProjectById`. Furthermore, activity one will return an error related to the activity. Maybe we could set this function as private and it won't appear in project and event

Reply:
I agree, just modified it